pkg/controller/jobs/deployment/deployment_webhook_test.go

```diff
@@ -62,20 +67,54 @@ func TestDefault(t *testing.T) {
 			deployment: testingdeployment.MakeDeployment("test-pod", "").PodTemplateSpecQueue("test-queue").Obj(),
 			want:       testingdeployment.MakeDeployment("test-pod", "").PodTemplateSpecQueue("test-queue").Obj(),
 		},
+		"LocalQueueDefaulting enabled, default lq is created, job doesn't have queue label": {
+			LocalQueueDefaulting: true,
+			defaultLqExist:       true,
+			deployment:           testingdeployment.MakeDeployment("test-pod", "default").Obj(),
+			want: testingdeployment.MakeDeployment("test-pod", "default").
+				Queue("default").
+				PodTemplateSpecQueue("default").
+				Obj(),
+		},
+		"LocalQueueDefaulting enabled, default lq is created, job has queue label": {
+			LocalQueueDefaulting: true,
+			defaultLqExist:       true,
+			deployment:           testingdeployment.MakeDeployment("test-pod", "").Queue("test-queue").Obj(),
+			want: testingdeployment.MakeDeployment("test-pod", "").
+				Queue("test-queue").
+				PodTemplateSpecQueue("test-queue").
+				Obj(),
+		},
+		"LocalQueueDefaulting enabled, default lq isn't created, job doesn't have queue label": {
+			LocalQueueDefaulting: true,
+			defaultLqExist:       false,
+			deployment:           testingdeployment.MakeDeployment("test-pod", "").Obj(),
+			want: testingdeployment.MakeDeployment("test-pod", "").
+				Obj(),
+		},
 	}
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
+			ctx, _ := utiltesting.ContextWithLog(t)
+			features.SetFeatureGateDuringTest(t, features.LocalQueueDefaulting, tc.LocalQueueDefaulting)
 			t.Cleanup(jobframework.EnableIntegrationsForTest(t, "pod"))
 			builder := utiltesting.NewClientBuilder()
-			client := builder.Build()
-
+			cli := builder.Build()
+			cqCache := cache.New(cli)
+			queueManager := queue.NewManager(cli, cqCache)
+			if tc.defaultLqExist {
+				if err := queueManager.AddLocalQueue(ctx, utiltesting.MakeLocalQueue("default", "default").
+					ClusterQueue("cluster-queue").
+					Obj()); err != nil {
+					t.Errorf("failed to create default local queue: %s", err)
```

Comment:
```suggestion
					t.Fatalf("failed to create default local queue: %s", err)
```
The same on other tests.